File: leetcode/455. Assign Cookies/main.rs
impl Solution {
    pub fn find_content_children(g: Vec<i32>, s: Vec<i32>) -> i32 {
        if s.is_empty() {
            return 0;
        }

        let mut g = g;
        let mut s = s;

        g.sort();
        s.sort();

        let mut max_num = 0;

        let mut cookie_idx = s.len() as i32 - 1;
        let mut child_idx = g.len() as i32 - 1;

        while cookie_idx >= 0 && child_idx >= 0 {
            if s[cookie_idx as usize] >= g[child_idx as usize] {
                max_num += 1;
                cookie_idx -= 1;
                child_idx -= 1;
            } else {
                child_idx -= 1;
            }
        }

        max_num
    }
}

File: leetcode/455. Assign Cookies/main.ts
const findContentChildren = (g: number[], s: number[]): number => {
  if (s.length == 0) return 0

  const sortable = (a: number, b: number) => a - b

  g.sort(sortable)
  s.sort(sortable)

  let maxNum = 0

  let cookieIdx = s.length - 1
  let childIdx = g.length - 1

  while (cookieIdx >= 0 && childIdx >= 0) {
    if (s[cookieIdx] >= g[childIdx]) {
      maxNum++

      cookieIdx--
      childIdx--
    } else {
      childIdx--
    }
  }

  return maxNum
}
